src/box/sql.c

```diff
@@ -1656,6 +1619,121 @@ sql_check_create(const char *name, uint32_t space_id, uint32_t func_id,
 	return sql_constraint_create(name, space_id, path, value);
 }
 
+const struct space *
+sql_space_by_token(const struct Token *name)
+{
+	char *name_str = sql_name_from_token(name);
+	struct space *res = space_by_name0(name_str);
+	sql_xfree(name_str);
+	return res;
+}
+
+uint32_t
+sql_index_id_by_token(const struct space *space, const struct Token *name)
+{
+	char *name_str = sql_name_from_token(name);
+	uint32_t res = UINT32_MAX;
+	for (uint32_t i = 0; i < space->index_count; ++i) {
+		if (strcmp(space->index[i]->def->name, name_str) == 0) {
+			res = space->index[i]->def->iid;
+			break;
+		}
+	}
+	sql_xfree(name_str);
+	return res;
+}
+
+uint32_t
+sql_fieldno_by_token(const struct space *space, const struct Token *name)
+{
+	char *name_str = sql_name_from_token(name);
+	uint32_t res = UINT32_MAX;
+	for (uint32_t i = 0; i < space->def->field_count; ++i) {
+		if (strcmp(space->def->fields[i].name, name_str) == 0) {
+			res = i;
+			break;
+		}
+	}
+	sql_xfree(name_str);
+	return res;
+}
+
+const struct tuple_constraint_def *
+sql_tuple_fk_by_token(const struct space *space, const struct Token *name)
+{
+	struct tuple_constraint_def *cdefs = space->def->opts.constraint_def;
+	uint32_t count = space->def->opts.constraint_count;
+	char *name_str = sql_name_from_token(name);
+	const struct tuple_constraint_def *res = NULL;
+	for (uint32_t i = 0; i < count; ++i) {
+		if (strcmp(cdefs[i].name, name_str) == 0 &&
+		    cdefs[i].type == CONSTR_FKEY) {
+			res = &cdefs[i];
+			break;
+		}
+	}
+	sql_xfree(name_str);
+	return res;
+}
+
+const struct tuple_constraint_def *
+sql_tuple_ck_by_token(const struct space *space, const struct Token *name)
+{
+	struct tuple_constraint_def *cdefs = space->def->opts.constraint_def;
+	uint32_t count = space->def->opts.constraint_count;
+	char *name_str = sql_name_from_token(name);
+	const struct tuple_constraint_def *res = NULL;
+	for (uint32_t i = 0; i < count; ++i) {
+		if (strcmp(cdefs[i].name, name_str) == 0 &&
+		    cdefs[i].type == CONSTR_FUNC) {
+			res = &cdefs[i];
+			break;
+		}
+	}
+	sql_xfree(name_str);
+	return res;
+}
+
+const struct tuple_constraint_def *
+sql_field_fk_by_token(const struct space *space, uint32_t fieldno,
+		      const struct Token *name)
```

Comment:
Minor: Why do you need to pass the token instead of `char *` here?

Reply:
I can pass `char *str` and `uint32_t len` here, but it will be inconvenient. It is also possible to pass normalized name here, but I  expand this function later for the second lookup.

Reply:
OK, thanks for explanation!